src/test/java/org/folio/rest/api/ScheduledNoticesAPITest.java

```diff
@@ -0,0 +1,308 @@
+    client.delete(scheduledNoticesStorageUrl("/scheduled-notices/" + noticeId), TENANT_ID, ResponseHandler.empty(deleteCompleted));
+    deleteCompleted.get(5, SECONDS);
+
+    CompletableFuture<Response> getCompleted = new CompletableFuture<>();
+    client.get(scheduledNoticesStorageUrl("/scheduled-notices/" + noticeId), TENANT_ID, ResponseHandler.empty(getCompleted));
+    Response response = getCompleted.get(5, SECONDS);
+
+    assertThat(response.getStatusCode(), is(404));
+  }
+
+  @Test
+  public void canDeleteAllScheduledNotices() throws InterruptedException, MalformedURLException, TimeoutException,
+    ExecutionException {
+
+    createScheduledNotice(NoticeConfig.Timing.BEFORE, ONE_DAY_PERIOD,
+      "a0d83326-d47e-43c6-8da6-f972015c3b52", NoticeConfig.Format.EMAIL);
+
+    createScheduledNotice(NoticeConfig.Timing.AFTER, ONE_MONTH_PERIOD,
+      "b84b721e-76ca-49ef-b486-f3debd92371d", NoticeConfig.Format.SMS);
+
+    CompletableFuture<Response> deleteCompleted = new CompletableFuture<>();
+    client.delete(scheduledNoticesStorageUrl("/scheduled-notices"), TENANT_ID, ResponseHandler.empty(deleteCompleted));
+    deleteCompleted.get(5, SECONDS);
+
+    CompletableFuture<JsonResponse> getCompleted = new CompletableFuture<>();
+    client.get(scheduledNoticesStorageUrl("/scheduled-notices"), TENANT_ID, ResponseHandler.json(getCompleted));
+    JsonResponse response = getCompleted.get(5, SECONDS);
+    ScheduledNotices scheduledNotices = response.getJson().mapTo(ScheduledNotices.class);
+
+    assertThat(scheduledNotices.getScheduledNotices().size(), is(0));
+    assertThat(scheduledNotices.getTotalRecords(), is(0));
+  }
+
+  @Test
+  public void canDeleteScheduledNoticesByQuery() throws Exception {
+
+    String templateId = "b84b721e-76ca-49ef-b486-f3debd92371d";
+
+    createScheduledNotice(NoticeConfig.Timing.BEFORE, ONE_DAY_PERIOD,
+      "a0d83326-d47e-43c6-8da6-f972015c3b52", NoticeConfig.Format.EMAIL);
+
+    createScheduledNotice(NoticeConfig.Timing.AFTER, ONE_DAY_PERIOD,
+      templateId, NoticeConfig.Format.SMS);
+
+    createScheduledNotice(NoticeConfig.Timing.UPON_AT, ONE_MONTH_PERIOD,
+      templateId, NoticeConfig.Format.SMS);
+
+    String query = "query=noticeConfig.templateId=" + templateId;
+
+    CompletableFuture<Response> deleteCompleted = new CompletableFuture<>();
+    client.delete(scheduledNoticesStorageUrl("/scheduled-notices?" + query), TENANT_ID, ResponseHandler.empty(deleteCompleted));
+    deleteCompleted.get(5, SECONDS);
+
+    CompletableFuture<JsonResponse> getCompleted = new CompletableFuture<>();
+    client.get(scheduledNoticesStorageUrl("/scheduled-notices?" + query), TENANT_ID, ResponseHandler.json(getCompleted));
+    JsonResponse response = getCompleted.get(5, SECONDS);
+
+    ScheduledNotices scheduledNotices = response.getJson().mapTo(ScheduledNotices.class);
+
+    assertThat(scheduledNotices.getScheduledNotices().size(), is(0));
```

Comment:
Hmm, this test appears to demonstrate deleting notices by `templateId`.

Two out of the three notices use that template, and the third record should still exist.

Why does it not check that the third notice still exists?

This test would pass even if all of the notices were deleted, which I imagine is not the desired behaviour.

Reply:
Added a check to verify that third notice still exists.